### PR TITLE
refactor Post::__get in order to simplify logic for menu items

### PR DESCRIPTION
**Pull request by vognev**
please check this pr, i think it somehow simplifies Post::__get() logic
also, tests cover as much as possible with current db dump

### CODE REVIEW

File: src/Post.php [hunk 1]
```diff
@@ -184,41 +184,45 @@ public function newQuery($excludeDeleted = true)
...
+
+        return $this->getAttribute($key);
+    }
+
+    /**
+     * Return object related to this nav_menu_item
+     *
+     * @return $this|\Illuminate\Database\Eloquent\Collection|\Illuminate\Database\Eloquent\Model
+     */
+    public function getMenuItemRelated()
```
Comment (jgrossi) [inline src/Post.php]: Should this method be inside `Menu` class and not `Post`?

File: src/PostMetaCollection.php [hunk 2]
```diff
@@ -29,6 +29,11 @@ public function __get($key)
         }
     }
 
+    public function __isset($key)
```
Comment (jgrossi) [inline src/PostMetaCollection.php]: Nice :-)

File: tests/PostTest.php [hunk 3]
```diff
@@ -176,4 +176,19 @@ public function testPostRelationConnections()
 
         $this->assertEquals('no_prefix', $post->author->getConnectionName());
     }
+
+    public function testMenuItemActsAsObjectFacadeForTitleAndSlug()
```
Comment (jgrossi) [inline tests/PostTest.php]: What type of menu testing are you making here? It's better to make menu tests inside `tests/MenuTest.php`.

File: tests/PostTest.php [hunk 4]
```diff
@@ -176,4 +176,19 @@ public function testPostRelationConnections()
...
+    {
+        $post = Post::find(16);
+        $this->assertEquals('Sample Page', $post->title);
+        $this->assertEquals('sample-page', $post->slug);
+
+        $post = Post::find(17);
+        $this->assertEquals('Hello world!', $post->title);
+        $this->assertEquals('hello-world', $post->slug);
+
+        $post = Post::find(18);
```
Comment (jgrossi) [inline tests/PostTest.php]: Please, check these posts ID. I think this records exist only inside your database, not the base one in `tests/config/database.sql`.

File: src/Post.php [hunk 5]
```diff
@@ -184,41 +184,45 @@ public function newQuery($excludeDeleted = true)
      */
     public function __get($key)
     {
-        if (!isset($this->$key)) {
-            if (isset($this->meta->$key)) {
-                return $this->meta->$key;
+        if (('nav_menu_item' == $this->getAttribute('post_type'))) {
```
Comment (jgrossi) [inline src/Post.php]: Why are you changing this order?

**Comment by jgrossi** [comment]
Thanks for the contribution @vognev :-) please, check my comments before merging this

**Comment by rundef** [comment]
Also, you might want to send the PR to the `dev` branch instead

**Comment by jgrossi** [comment]
Yes! Thanks @rundef I forgot that 😁

**Pull request closed without merging**